generators/server/templates/src/main/java/package/security/SecurityUtils.java.ejs

```diff
@@ -59,6 +65,13 @@ public final class SecurityUtils {
                 if (authentication.getPrincipal() instanceof UserDetails) {
                     UserDetails springSecurityUser = (UserDetails) authentication.getPrincipal();
                     return springSecurityUser.getUsername();
+                <%_ if (authenticationType === 'oauth2') { _%>
+                } else if (authentication.getPrincipal() instanceof DefaultOidcUser) {
+                    Map<String, Object> attributes = ((DefaultOidcUser) authentication.getPrincipal()).getAttributes();
+                    if (attributes.containsKey("preferred_username")) {
```

Comment:
Should we put "preferred_username" in a constant field or at least in local variable?

Reply:
according to @pascalgrimaud we should create constants for oauth2 fields, obviously in another PR.